lib/provider/default_volume_provider_test.go

```diff
@@ -0,0 +1,185 @@
+/**
+ * Copyright 2020 IBM Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// Package provider ...
+package provider
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"go.uber.org/zap"
+)
+
+var (
+	logger *zap.Logger
+)
+
+func init() {
+	logger, _ = zap.NewDevelopment()
+}
+
+func TestForDefaultVolumeProviderTypeAndName(t *testing.T) {
+	ccf := &DefaultVolumeProvider{sess: nil}
+
+	volumeProvider := ccf.ProviderName()
+
+	volumeType := ccf.Type()
+
+	volumeProviderName := ccf.GetProviderDisplayName()
+
+	assert.Empty(t, volumeProviderName)
+	assert.Empty(t, volumeProvider)
+	assert.Empty(t, volumeType)
+}
+
+func TestForCreateVolume(t *testing.T) {
+	ccf := &DefaultVolumeProvider{sess: nil}
+
+	providerVolume :=
+		Volume{
+			VolumeID: "16f293bf-test-4bff-816f-e199c0c65db5",
+			Name:     String("test volume name"),
+			Capacity: nil,
+		}
+	volume, _ := ccf.CreateVolume(providerVolume)
+	assert.Nil(t, volume)
+}
+
+func TestForAttachDetachVolume(t *testing.T) {
+	ccf := &DefaultVolumeProvider{sess: nil}
+
+	attachRequest :=
```

Comment:
I don't think so you need to create proper object, I was thinking to do it following way 

assert.Nil(t, ccf.AttachVolume(VolumeAttachmentRequest{}))

similar for other as well

Reply:
We can avoid declaration of dummy object but we need variable for this as it return 2 values and assert.Nil wont allow 
We can do something as below.
func TestForAttachVolume(t *testing.T) {
	ccf := &DefaultVolumeProvider{sess: nil}

	volume, _ := ccf.AttachVolume(VolumeAttachmentRequest{})
	assert.Nil(t, volume)
}